helix-term/src/ui/editor.rs

```diff
@@ -1169,19 +1178,28 @@ impl EditorView {
 
                 let (view, doc) = current!(cxt.editor);
 
-                if doc
-                    .selection(view.id)
-                    .primary()
-                    .slice(doc.text().slice(..))
-                    .len_chars()
-                    <= 1
-                {
-                    return EventResult::Ignored(None);
-                }
+                let should_yank = match cxt.editor.mouse_down_range {
```

Comment:
```suggestion
                let should_yank = match cxt.editor.mouse_down_range.take() {
```

and then the `cxt.editor.mouse_down_range = None` can be eliminated below

Reply:
Ah TIL that `Option::take()` exists. Thanks! It's been fixed.